frontend/src/components/grantaccess/GrantApplicationAdmin.vue

```diff
@@ -0,0 +1,188 @@
+import { ErrorMessage, Field, Form as VeeForm } from 'vee-validate';
+import { object, string } from 'yup';
+import router from '@/router';
+import Dropdown from 'primevue/dropdown';
+import { AdminMgmtApiService } from '@/services/ApiServiceFactory';
+import { UserType, type FamApplication } from 'fam-app-acsctl-api';
+import type { FamAppAdminCreateRequest } from 'fam-admin-mgmt-api/model/fam-app-admin-create-request';
+import type { FamApplicationGetResponse } from 'fam-admin-mgmt-api/model/fam-application-get-response';
+import Button from '@/components/common/Button.vue';
+import { IconSize } from '@/enum/IconEnum';
+import { Severity } from '@/enum/SeverityEnum';
+
+import { isLoading } from '@/store/LoadingState';
+import { setNotificationMsg } from '@/store/NotificationState';
+
+const defaultFormData = {
+    userId: '',
+    application: Number,
+};
+const formData = ref(JSON.parse(JSON.stringify(defaultFormData))); // clone default input
+const formValidationSchema = object({
+    userId: string()
+        .required('User ID is required')
+        .min(2, 'User ID must be at least 2 characters')
+        .nullable(),
+    application: object().required('Application is required'),
+});
+
+const applications = ref<FamApplicationGetResponse[]>();
+
+// This is going to be changed when the new backend API is ready
+onMounted(async () => {
+    applications.value = (
+        await AdminMgmtApiService.applicationsApi.getApplications()
+    ).data;
+});
+
+/* ------------------ User information method ------------------------- */
+const userIdChange = (userId: string) => {
+    formData.value.userId = userId;
+};
+
+const verifyUserIdPassed = ref(false);
+const setVerifyUserIdPassed = (verifiedResult: boolean) => {
+    verifyUserIdPassed.value = verifiedResult;
+};
+/* ---------------------- Form method ---------------------------------- */
+const cancelForm = () => {
+    formData.value = defaultFormData;
+    router.push('/dashboard');
+};
+
+const toRequestPayload = (formData: any) => {
+    const request = {
+        user_name: formData.userId,
+        application_id: formData.application.application_id,
+        user_type_code: UserType.I,
+    } as FamAppAdminCreateRequest;
+    return request;
+};
+
+const handleSubmit = async () => {
+    const data = toRequestPayload(formData.value);
+    await AdminMgmtApiService.applicationAdminApi
+        .createApplicationAdmin(data)
+        .then(() => {
+            setNotificationMsg(
+                Severity.success,
+                `Admin privilege has been added to ${formData.value.userId.toUpperCase()} for application ${
+                    formData.value.application.application_name
+                }`
+            );
+        })
+        .catch((error) => {
+            if (error.response?.status === 409) {
+                setNotificationMsg(
+                    Severity.warning,
+                    error.response.data.detail
+                );
```

Comment:
For now, maybe it is ok, but this logic inside catch() block look strange: on this `409` case and on `else { }` block at line 90.
If the backend returns errors and are generic enough for frontend and does not need to be processed, frontend components are not necessary to handle them individually.

Reply:
I agree, maybe this could be a conversation to improve the error notification handling.